pkg/ccl/backupccl/testdata/backup-restore/restore-on-fail-or-cancel-ttl

```diff
@@ -0,0 +1,70 @@
+new-server name=s1 nodes=1
+----
+
+subtest restore-fail-cleans-up-ttl-schedules-before-publishing
+
+exec-sql
+CREATE DATABASE d;
+CREATE TABLE d.tb (id INT PRIMARY KEY) WITH (ttl_expire_after = '10 minutes');
+----
+
+exec-sql
+BACKUP DATABASE d INTO 'userfile:///foo'
+----
+
+# Attempt the restore but pause it before publishing descriptors.
+exec-sql
+SET CLUSTER SETTING jobs.debug.pausepoints = 'restore.before_publishing_descriptors';
+DROP DATABASE d;
+----
+
+restore expect-pausepoint tag=a
+RESTORE DATABASE d FROM LATEST IN 'userfile:///foo';
+----
+job paused at pausepoint
+
+# Cancel the job so that the cleanup hook runs.
+job cancel=a
+----
+
+query-sql
+SELECT count(1) FROM [SHOW SCHEDULES] WHERE label LIKE 'row-level-ttl-%';
```

Comment:
@adityamaru could you add a note at the top of the test to explain the gist of this test? "Any table created with the `ttl_expire_after` parameter creates a scheduled ttl job. So, if a restore of this table's database gets canceled, we expect this ttl job to not exist in the schedule table"